src/writeData/components/PluginAddToExistingConfiguration/Footer.tsx

```diff
@@ -0,0 +1,207 @@
+          confirmationButtonColor={ComponentColor.Primary}
+          confirmationLabel={`This configuration already contains ${pluginConfigName} Input Plugin. Are you sure you want to add another?`}
+          status={ComponentStatus.Valid}
+          tabIndex={0}
+          testID="plugin-add-to-configuration"
+        />
+      ) : (
+        <Button
+          color={ComponentColor.Primary}
+          text="Add to Existing Config"
+          onClick={onConfirm}
+          status={
+            isValidConfiguration
+              ? ComponentStatus.Valid
+              : ComponentStatus.Disabled
+          }
+          tabIndex={0}
+          testID="plugin-add-to-configuration--button"
+        />
+      )}
+    </>
+  )
+}
+
+const FooterCustomize: FC<FooterProps> = props => {
+  const {onCancel, onConfirm, isValidConfiguration} = props
+
+  return (
+    <>
+      <Button
+        color={ComponentColor.Default}
+        onClick={onCancel}
+        tabIndex={1}
+        testID="plugin-add-to-configuration-previous"
+        text="Previous"
+      />
+      {isValidConfiguration ? (
+        <ConfirmationButton
+          color={ComponentColor.Primary}
+          text="Save and Test"
+          onConfirm={onConfirm}
+          confirmationButtonText="Save"
+          confirmationButtonColor={ComponentColor.Primary}
+          confirmationLabel="Are you sure you want to save changes?"
+          status={ComponentStatus.Valid}
+          tabIndex={0}
+          testID="plugin-add-to-configuration-save-and-test"
+        />
+      ) : (
+        <Button
+          color={ComponentColor.Primary}
+          text="Save and Test"
+          onClick={onConfirm}
+          status={ComponentStatus.Disabled}
+          tabIndex={0}
+          testID="plugin-add-to-configuration-save-and-test--button"
+        />
+      )}
+    </>
+  )
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = PluginConfigurationStepProps & ReduxProps
+
+const FooterComponent: FC<Props> = props => {
+  const {
+    currentStepIndex,
+    isValidConfiguration,
+    onDecrementCurrentStepIndex,
+    onExit,
+    onIncrementCurrentStepIndex,
+    onUpdateTelegraf,
+    pluginConfig,
+    pluginConfigName,
+    setIsValidConfiguration,
+    telegrafConfig,
+    telegrafConfigDescription,
+    telegrafConfigName,
+  } = props
+
+  const handleContinueFromConfigure = () => {
+    setIsValidConfiguration(false)
+    onIncrementCurrentStepIndex()
+  }
+
+  const handleSaveAndTest = () => {
+    onUpdateTelegraf({
+      ...telegrafConfig,
+      config: pluginConfig,
+      description: telegrafConfigDescription,
+      name: telegrafConfigName,
+    })
+    onIncrementCurrentStepIndex()
+  }
+
+  const pattern = `[[inputs.${pluginConfigName}]]`
+
+  let FooterButtons = null
+
+  if (currentStepIndex === PluginConfigurationStep.Configure) {
+    FooterButtons = (
+      <FooterConfigure
+        hasDuplicatePlugin={telegrafConfig?.config?.includes(pattern) ?? false}
```

Comment:
Just be aware that `??` only fails on `null` and `undefined`. It seems like that's what we want here, but I just wanted to call that out

Reply:
Yes, completely intended here. If any property is missing (`undefined`) or if the pattern can't be found (`undefined`), there are no duplicates (`false`). `null` can also substitute for the `undefined` if a property is deliberately nothing.